node_modules/col-assets/tests/test-categories.js

```diff
@@ -0,0 +1,251 @@
+
+                          // Duplicate category id
+                          opts = {'categories': [category.id, category.id]};
+                          AssetsTestUtil.assertCreateLinkBookmarkletFails(bookmarkletClient, course, me.id, me.bookmarklet_token, 'UC Davis', 'http://www.ucdavis.edu/', opts, 400, function() {
+
+                            return callback();
+                          });
+                        });
+                      });
+                    });
+                  });
+                });
+              });
+            });
+          });
+        });
+      });
+
+      /**
+       * Test that verifies authorization when create a new asset with associated categories
+       */
+      it('verifies authorization', function(callback) {
+        var adminUser1 = TestsUtil.generateAdminUser();
+        TestsUtil.getAssetLibraryClient(null, null, adminUser1, function(client1, course1, adminUser1) {
+          CategoriesTestUtil.assertCreateCategory(client1, course1, 'Category 1', function(category) {
+
+            // Verify that a category in a different course can not be used
+            TestsUtil.getAssetLibraryClient(null, null, null, function(client2, course2, user2) {
+              UsersTestUtil.assertGetMe(client2, course2, null, function(me) {
+                var bookmarkletClient = TestsUtil.getAnonymousClient();
+
+                var opts = {'categories': category.id};
+                AssetsTestUtil.assertCreateLinkBookmarkletFails(bookmarkletClient, course2, me.id, me.bookmarklet_token, 'UC Davis', 'http://www.ucdavis.edu/', opts, 404, function() {
+
+                  return callback();
+                });
+              });
+            });
+          });
+        });
+      });
+    });
+
+    describe('Get asset', function() {
+
+      /**
+       * Test that verifies that the categories are included when retrieving an asset
+       */
+      it('can be retrieved', function(callback) {
+        var adminUser = TestsUtil.generateAdminUser();
+        TestsUtil.getAssetLibraryClient(null, null, adminUser, function(client, course, adminUser) {
+          CategoriesTestUtil.assertCreateCategory(client, course, 'Category 1', function(category1) {
+            CategoriesTestUtil.assertCreateCategory(client, course, 'Category 2', function(category2) {
+
+              var opts = {'categories': [category1.id, category2.id]};
+              AssetsTestUtil.assertCreateLink(client, course, 'UC Berkeley', 'http://www.berkeley.edu/', opts, function(asset) {
+                CategoriesTestUtil.assertCategory(asset.categories[0], category1);
+                CategoriesTestUtil.assertCategory(asset.categories[1], category2);
+
+                // Ensure that the categories are returned when getting the asset
+                AssetsTestUtil.assertGetAsset(client, course, asset.id, asset, 0, function(asset) {
```

Comment:
It does actually. When creating the link, we assert that both categories are present. When getting the link, we pass in `asset` as the expected asset, which will compare the categories as well. I can add an explicit check here if you want though.